cmd/dm-worker/main_test.go

```diff
@@ -32,6 +40,25 @@ func TestRunMain(t *testing.T) {
 		}
 	}
 
+	oldOsExit := utils.OsExit
+	defer func() { utils.OsExit = oldOsExit }()
+	utils.OsExit = func(code int) {
+		log.Infof("[test] os.Exit with code %d", code)
+		exit <- code
+		// sleep here to prevent following code execution in the caller routine
+		time.Sleep(time.Second * 60)
+	}
+
 	os.Args = args
-	main()
+	go func() {
+		main()
+		close(waitCh)
+	}()
+
+	select {
+	case <-waitCh:
+		return
+	case <-exit:
+		return
```

Comment:
so the method is to exit from this _main_ routine to terminal the separate `main()` routine (and all other routines)?

Reply:
yes. (to achieve the goal-2, `main` routine must exit normally)